analyzers/src/SonarAnalyzer.CSharp/Rules/BlazorQueryParameterTypeShouldBeSupported.cs

```diff
@@ -0,0 +1,106 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class BlazorQueryParameterTypeShouldBeSupported : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6797";
+    private const string MessageFormat = "Query parameter type '{0}' is not supported.";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    private static readonly ISet<KnownType> SupportedTypes = new HashSet<KnownType>
+    {
+        KnownType.System_Boolean,
+        KnownType.System_DateTime,
+        KnownType.System_Decimal,
+        KnownType.System_Double,
+        KnownType.System_Single,
+        KnownType.System_Int32,
+        KnownType.System_Int64,
+        KnownType.System_String,
+        KnownType.System_Guid
+    };
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(cc =>
+        {
+            if (cc.Compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Components_RouteAttribute) is null)
+            {
+                return;
+            }
+
+            cc.RegisterSymbolAction(c =>
+            {
+                var componentClass = (INamedTypeSymbol)c.Symbol;
+                if (HasRouteAttribute(componentClass))
+                {
+                    foreach (var propertyType in GetPropertyTypeMismatches(componentClass))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, propertyType.GetLocation(), GetTypeName(propertyType)));
+                    }
+                }
+            },
+            SymbolKind.NamedType);
+        });
+
+    private static string GetTypeName(TypeSyntax propertyType)
+    {
+        if (propertyType.NameIs(KnownType.System_Nullable_T.TypeName))
+        {
+            propertyType = ((GenericNameSyntax)propertyType).TypeArgumentList.Arguments[0];
+        }
+
+        return propertyType.GetName();
+    }
+
+    private static IEnumerable<TypeSyntax> GetPropertyTypeMismatches(INamedTypeSymbol componentClass) =>
+        componentClass
+            .GetMembers()
+            .Where(IsPropertyTypeMismatch)
+            .SelectMany(p => p.DeclaringSyntaxReferences.Select(d => ((PropertyDeclarationSyntax)d.GetSyntax()).Type));
+
+    private static bool IsPropertyTypeMismatch(ISymbol member) =>
+        member is IPropertySymbol property
+        && property.HasAttribute(KnownType.Microsoft_AspNetCore_Components_SupplyParameterFromQueryAttribute)
+        && !SupportedTypes.Any(t => IsSupportedType(property.Type, t));
```

Comment:
Single variable lambdas should use `x` as the variable name

Reply:
Please review the whole file, there are more instances where you should rename the variables 👍